cppcache/src/PdxInstanceImpl.cpp

```diff
@@ -1854,12 +1823,12 @@ void PdxInstanceImpl::setField(const std::string& fieldName, bool* value,
                                 " or type of field not matched " +
                                 (pft != nullptr ? pft->toString() : ""));
   }
-  auto cacheableObject = BooleanArray::create(value, length);
+  auto cacheableObject = CacheableArray<bool>::create(value);
   m_updatedFields[fieldName] = cacheableObject;
 }
 
-void PdxInstanceImpl::setField(const std::string& fieldName, int8_t* value,
-                               int32_t length) {
+void PdxInstanceImpl::setField(const std::string& fieldName,
+                               std::vector<int8_t> value) {
   auto pt = getPdxType();
   auto pft = pt->getPdxField(fieldName.c_str());
```

Comment:
Can you fix all these while in here. I don't think `.c_str()` is required anymore.